File: katas/6 kyu/write-number-in-expanded-form/solution.php
<!-- write-number-in-expanded-form -->
<?php
use PHPUnit\Framework\TestCase;
/* ------------------------------- MY SOLUTION ------------------------------ */
function expanded_form(int $n): string {
  $split = str_split($n);
  $num_digits = count($split);
  $numbers_arr = [];
  foreach ($split as $digit) {
    if ($digit != 0) {
      $numbers_arr[] = $digit . str_repeat(0, $num_digits - 1);
    }
    $num_digits--;
  }
  return implode(' + ', $numbers_arr);
}
/* ----------------------------- CLEVER SOLUTION ---------------------------- */
/* ---------------------------------- TEST ---------------------------------- */
class ExpandedFormTest extends TestCase {
  public function testDescriptionExamples() {
    $this->assertSame("10 + 2", expanded_form(12));
    $this->assertSame("40 + 2", expanded_form(42));
    $this->assertSame("70000 + 300 + 4", expanded_form(70304));
  }
}

File: katas/6 kyu/write-number-in-expanded-form/solution.py
# write-number-in-expanded-form
import codewars_test as test
# -------------------------------- MY SOLUTION ------------------------------- #
def expanded_form(num):
  result = []

  for index, digit in enumerate(str(num)[::-1]):
    if int(digit) != 0:
      result.append(digit + ('0' * index))

  return ' + '.join(result[::-1])
# ------------------------------ CLEVER SOLUTION ----------------------------- #
def expanded_form(num):
  num = list(str(num))
  return ' + '.join(x + '0' * (len(num) - y - 1) for y,x in enumerate(num) if x != '0')
# ----------------------------------- TEST ----------------------------------- #
test.assert_equals(expanded_form(12), '10 + 2');
test.assert_equals(expanded_form(42), '40 + 2');
test.assert_equals(expanded_form(70304), '70000 + 300 + 4');